tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/SQS/ContextPropagation.cs

```diff
@@ -33,6 +35,28 @@ public static void InjectHeadersIntoMessage<TMessageRequest>(IContainsMessageAtt
             {
                 carrier.MessageAttributes = CachedMessageHeadersHelper<TMessageRequest>.CreateMessageAttributes();
             }
+            else
+            {
+                List<string> attributesToRemove = null;
+
+                // Make sure we do not propagate any other datadog header here in the rare cases where users would have added them manually
+                foreach (var attribute in carrier.MessageAttributes.Keys)
+                {
+                    if (attribute is string attributeName && attributeName.StartsWith("x-datadog", StringComparison.OrdinalIgnoreCase))
+                    {
+                        attributesToRemove ??= new List<string>();
+                        attributesToRemove.Add(attributeName);
+                    }
+                }
+
+                if (attributesToRemove != null)
+                {
+                    foreach (var attribute in attributesToRemove)
+                    {
+                        carrier.MessageAttributes.Remove(attribute);
+                    }
+                }
```

Comment:
I think this is another case where the extra list and copy is not necessary. It's just `MessageAttributes` is a `Dictionary<,>`, and (as far as I know), it's safe to enumerate the keys while removing from the dictionary.

Reply:
I'll check thanks

Reply:
So indeed, removing will work. Adding will throw.:til: Thanks a lot.
Fixed in b271dbd18